frontend/appflowy_flutter/lib/startup/tasks/generate_router.dart

```diff
@@ -0,0 +1,270 @@
+import 'package:appflowy/mobile/presentation/presentation.dart';
+import 'package:appflowy/startup/startup.dart';
+import 'package:appflowy/startup/tasks/app_widget.dart';
+import 'package:appflowy/user/application/auth/auth_service.dart';
+import 'package:appflowy/user/presentation/presentation.dart';
+import 'package:appflowy/workspace/presentation/home/desktop_home_screen.dart';
+import 'package:flowy_infra/time/duration.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
+
+GoRouter generateRouter(Widget child) {
+  return GoRouter(
+    navigatorKey: AppGlobals.rootNavKey,
+    initialLocation: '/',
+    routes: [
+      GoRoute(
+        path: '/',
+        // Root route is SplashScreen.
+        // It needs LaunchConfiguration as a parameter, so we get it from ApplicationWidget's child.
+        pageBuilder: (context, state) => MaterialPage(
+          child: child,
+        ),
+      ),
+      // Routes in both desktop and mobile
+      GoRoute(
+        path: SignInScreen.routeName,
+        pageBuilder: (context, state) {
+          return CustomTransitionPage(
+            child: SignInScreen(router: getIt<AuthRouter>()),
+            transitionsBuilder: _buildFadeTransition,
+            transitionDuration: Duration(
```

Comment:
"Define the duration as a constant value, and then you can reuse it in the code below.

```dart
            transitionDuration: _transitionDuration,

// ...


final _transitionDuration = Duration(
  milliseconds: (RouteDurations.slow.inMilliseconds).round(),
);

```